volatility3/framework/layers/intel.py

```diff
@@ -338,33 +371,19 @@ def _page_is_valid(entry: int) -> bool:
         return bool((entry & 1) or ((entry & 1 << 11) and not entry & 1 << 10))
 
     def _handle_page_fault(self, layer_name, offset, invalid_bits, entry, description):
+        kernel = self._get_kernel_module()
```

Comment:
This should be memoized, otherwise there's going to be (potentially) a bunch of page fails that retraverse the module list to find the kernel.  I'd consider a `_kernel` property which calculates it first time, and returns it from a variable otherwise.

Reply:
moved `self._kernel` initialization to `Intel` constructor.